File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<string> stk;

        for(auto& str:tokens){
            if(str=="*" || str=="/" || str=="+" || str=="-"){
                int a = stoi(stk.top());
                stk.pop();
                int b = stoi(stk.top());
                stk.pop();

                int ans = 0;
                
                if(str=="*") ans += (a*b);
                else if(str == "/") ans += (b/a);
                else if(str == "+") ans += (a+b);
                else ans += (b-a);

                stk.push(to_string(ans));
            }
            else stk.push(str);
        }
        return stoi(stk.top());
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for char in tokens:
            if char in '+-/*':
                a = int(stack.pop())
                b = int(stack.pop())
                c = 0
                if char == '+':
                    c = a+b
                elif char == '-':
                    c = b-a
                elif char == '*':
                    c = b*a
                else:
                    c = b/a

                stack.append(c)
                
            else:
                stack.append(char)
            
        
        return int(stack.pop())

        